executor/src/witgen/bit_constraints.rs

```diff
@@ -52,7 +56,9 @@ impl BitConstraint {
     /// The bit constraint of an integer multiple of an expression.
     /// TODO this assumes goldilocks
     pub fn multiple(&self, factor: FieldElement) -> Option<BitConstraint> {
-        if factor.to_integer() * self.mask >= FieldElement::modulus() {
+        if factor.to_arbitrary_integer() * self.mask.into_biguint()
```

Comment:
@chriseth could you explain this line? I'm a bit surprised to see a multiplication by a mask

Reply:
Hm yeah, interesting. This of course only works if `factor` is a power of two. Is that checked outside?

Reply:
So what's the conclusion here?

Reply:
It's not clear to me that `factor` is always a power of two, as it is a coefficient in an arbitrary `AffineExpression`.

Reply:
I would approve but don't know what needs to be done here

Reply:
I moved this to #270